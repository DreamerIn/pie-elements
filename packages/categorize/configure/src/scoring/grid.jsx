import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import classNames from 'classnames';
import grey from '@material-ui/core/colors/grey';

export const Cell = withStyles(theme => ({
  cell: {
    width: '100%',
    height: '100%',
    backgroundColor: 'white',
    alignSelf: 'center',
    borderBottom: `solid 1px ${grey[300]}`,
    display: 'flex',
    alignItems: 'center'
  }
}))(
  ({ classes, children, html }) =>
    html ? (
      <div
        className={classes.cell}
        dangerouslySetInnerHTML={{ __html: html }}
      />
    ) : (
      <div className={classes.cell}>{children}</div>
    )
);

Cell.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]),
  html: PropTypes.string
};
export const Heading = withStyles(theme => ({
  heading: {
    fontWeight: 'bold',
    paddingBottom: 0,
    marginBottom: theme.spacing.unit
  }
}))(({ classes, children }) => (
  <Cell>
    <div className={classes.heading}>{children}</div>
  </Cell>
));

export class Grid extends React.Component {
  static propTypes = {
    classes: PropTypes.object.isRequired,
    className: PropTypes.string,
    children: PropTypes.oneOfType([
      PropTypes.arrayOf(PropTypes.node),
      PropTypes.node
    ]).isRequired,
    headings: PropTypes.arrayOf(PropTypes.string),
    size: PropTypes.string
  };

  static defaultProps = {};
  render() {
    const { classes, className, children, headings, size } = this.props;
    const length = Array.isArray(headings) ? headings.length : 1;
    const columns = size ? size : `repeat(${length}, 1fr)`;
    const style = {
      gridTemplateColumns: columns
    };

    return (
      <div style={style} className={classNames(classes.grid, className)}>
        {headings.map((h, index) => <Heading key={index}>{h}</Heading>)}
        {children}
      </div>
    );
  }
}
const styles = () => ({
  grid: {
    display: 'grid',
    backgroundColor: 'red',
    gridRowGap: '1px',
    gridColumnGap: 0
  }
});
export default withStyles(styles)(Grid);
