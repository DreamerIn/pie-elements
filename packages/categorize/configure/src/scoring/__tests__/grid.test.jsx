import { shallow } from 'enzyme';
import React from 'react';

import { Grid } from '../grid';

describe('Grid', () => {
  let w;
  let onChange = jest.fn();
  const wrapper = extras => {
    const defaults = {
      classes: {},
      className: 'className',
      onChange,
      headings: ['A', 'B']
    };
    const props = { ...defaults, ...extras };
    return shallow(
      <Grid {...props}>
        <div>a</div>
        <div>b</div>
      </Grid>
    );
  };
  describe('snapshot', () => {
    it('renders', () => {
      w = wrapper();
      expect(w).toMatchSnapshot();
    });
  });
});
