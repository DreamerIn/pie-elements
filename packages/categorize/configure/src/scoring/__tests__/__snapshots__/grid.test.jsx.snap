// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid snapshot renders 1`] = `
<div
  className="className"
  style={
    Object {
      "gridTemplateColumns": "repeat(2, 1fr)",
    }
  }
>
  <WithStyles(Component)
    key="0"
  >
    A
  </WithStyles(Component)>
  <WithStyles(Component)
    key="1"
  >
    B
  </WithStyles(Component)>
  <div>
    a
  </div>
  <div>
    b
  </div>
</div>
`;
